src/integrationTest/java/uk/gov/hmcts/reform/userprofileapi/integration/AddRolesWithIdamIntTest.java

```diff
@@ -246,4 +276,149 @@ public void should_return_400_and_not_create_user_profile_when_empty_body_delete
 
     }
 
+
+    @Test
+    public void should_see_error_message_from_idam_when_role_addition_fails_and_sidam_returns_error_response() throws Exception {
+
+        UserProfile userProfile = buildUserProfile();
+        userProfile.setStatus(IdamStatus.ACTIVE);
+        UserProfile persistedUserProfile = userProfileRepository.save(userProfile);
+
+        String userId = persistedUserProfile.getIdamId();
+        assertThat(userId).isNotNull();
+
+        mockWithUpdateRolesFailure(HttpStatus.PRECONDITION_FAILED, true, userId);
+
+        UpdateUserProfileData userRoles = new UpdateUserProfileData();
+
+        RoleName role1 = new RoleName("puicase-manager");
+        RoleName role2 = new RoleName("prd-Admin");
+

```

Comment:
please move the code to one block and reuse the code in all test cases. i am seeing lot of duplication of the code

Reply:
done